addons/event/models/mail_template.py

```diff
@@ -20,3 +20,10 @@ def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_ui
         if self.env.context.get('filter_template_on_event'):
             args = expression.AND([[('model', '=', 'event.registration')], args])
         return super(MailTemplate, self)._name_search(name, args, operator, limit, name_get_uid)
+
+    def unlink(self):
+        res = super(MailTemplate, self).unlink()
+        domain = ('template_ref', 'in', [f"{template._name},{template.id}" for template in self])
+        for model in ['event.mail', 'event.type.mail']:
+            self.env[model].sudo().search([domain]).unlink()
```

Comment:
I would prefer two explicit search/unlink (self.env['event.mail'] and self.env['event.type.mail']), as we are not going to add a new model, hence the loop itself does not add much.